state/core_access.go

```diff
@@ -158,17 +156,22 @@ func (ca *CoreAccessor) constructSignedTx(
 
 func (ca *CoreAccessor) SubmitPayForBlob(
 	ctx context.Context,
-	nID namespace.ID,
-	data []byte,
 	fee Int,
 	gasLim uint64,
+	blobs ...*apptypes.Blob,
 ) (*TxResponse, error) {
-	b := &apptypes.Blob{NamespaceId: nID, Data: data, ShareVersion: uint32(appconsts.DefaultShareVersion)}
+	if len(blobs) == 0 {
+		return nil, errors.New("state: no blobs provided")
+	}
+
+	b := make([]*apptypes.Blob, len(blobs))
+	copy(b, blobs)
```

Comment:
Did you mean to use b later to protect `blobs` from modifications? There is something wring, since b is not used after copy.